### PR TITLE
RUB-138 related changes

### PR DESCRIPTION
**Pull request by ishwari-gaikwad2**

### CODE REVIEW

File: force-app/main/default/classes/OneJNEventService.cls [hunk 1]
```diff
@@ -584,4 +642,38 @@ public with sharing class OneJNEventService {
       return oneJNDeserializerVar;
     }
   }
+
+  public static List<Document_Instruction__c> getDocumentInstructionInstance(
```
Comment (RemarioCRM) [inline force-app/main/default/classes/OneJNEventService.cls:646]: add jsdocs

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/OneJNDeserializer.cls [hunk 2]
```diff
@@ -103,10 +104,11 @@ public class OneJNDeserializer {
     @AuraEnabled
     public decimal TotalFees;
   }
+    /*
   public class DocumentRequirementsWrapper {
```
Comment (RemarioCRM) [inline force-app/main/default/classes/OneJNDeserializer.cls:108]: delete dead code

File: force-app/main/default/classes/OneJNSerializer.cls [hunk 3]
```diff
@@ -153,7 +153,24 @@ public class OneJNSerializer {
         this.ApplicationId = applicationRecord.Application_ID__c;
         this.ApprovedTenure = applicationRecord.IDM_Tenure_Date__c;
         this.ApprovedCreditAmount = applicationRecord.IDM_Loan_Amount_Number__c;
-        this.AdjudicationDecision = applicationRecord.Adjudication_Decision_picklist__c;
+
+        /*
```
Comment (RemarioCRM) [inline force-app/main/default/classes/OneJNSerializer.cls:157]: remove dead code

File: force-app/main/default/classes/OneJNSerializer.cls [hunk 4]
```diff
@@ -153,7 +153,24 @@ public class OneJNSerializer {
...
+        if( applicationRecord.Adjudication_Decision_picklist__c ==Constants.MANUALLY_APPROVED
+        || applicationRecord.Adjudication_Decision_picklist__c ==Constants.DEFERRAL_APPROVED ){
+          this.AdjudicationDecision = Constants.MANUAL_DECISION_ACCEPT;
+        }else if(applicationRecord.Adjudication_Decision_picklist__c ==Constants.MANUALLY_DECLINED
+        || applicationRecord.Adjudication_Decision_picklist__c ==Constants.DEFERRAL_DECLINED){
+          this.AdjudicationDecision = Constants.MANUAL_DECISION_DECLINE;
+        }*/
+
+        if( applicationRecord.Adjudication_Decision_picklist__c ==Constants.MANUAL_APPROVE
```
Comment (RemarioCRM) [inline force-app/main/default/classes/OneJNSerializer.cls:166]: what is this change for?

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/OneJNSerializer.cls (hunk 4 above)
Comment (ishwari-gaikwad2) [reply force-app/main/default/classes/OneJNSerializer.cls:166]: This is for RUB-656 picklist values related changes

File: force-app/main/default/classes/OneJNEventService.cls [hunk 5]
```diff
@@ -131,10 +131,25 @@ public with sharing class OneJNEventService {
...
+      List<String> documentRequirementList = oneJNDeserializerVar.ConsolidatedReport
+        ?.ApplicantDetailsPrimary
+        ?.DocumentRequirements;
+      DocumentChecklistItem documentChecklistItem = new DocumentChecklistItem();
+      if (
+        documentRequirementList <> null &&
+        documentRequirementList.size() > 0
+      ) {
+        documentChecklistItem = getDocumentCheckListItem(opportunity1[0]);
+        insert documentChecklistItem;
```
Comment (RemarioCRM) [inline force-app/main/default/classes/OneJNEventService.cls:151]: SobjectDecision class for security.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/OneJNEventService.cls (hunk 5 above)
Comment (ishwari-gaikwad2) [reply force-app/main/default/classes/OneJNEventService.cls:151]: @RemarioCRM - as we are creating document checklist items from the automated process. when i add sObjectDecision class it gives an error that "No access to entity: DocumentChecklistItem". so i tried to assign document checklist item permission set to that user but not able to do that.. what can be done in this case?

Comment (ishwari-gaikwad2) [reply force-app/main/default/classes/OneJNEventService.cls:151]: @RemarioCRM - when i add the sObjectDecision for insertion of the Document checklist item i am getting this error: "No access to entity: DocumentChecklistItem". I think this is due to the record is getting created from the Automated process.
so i tried assigning the "DocumentChecklistItem" permission set to that user but unable to do that.
Please suggest.

Comment (RemarioCRM) [reply force-app/main/default/classes/OneJNEventService.cls:151]: @ishwari-gaikwad2 the class is invoked from a queueable from a flow as described previously, so it's not an automated process.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

**Review by RemarioCRM — APPROVED** [review]

**Pull request merged**